cyclops-ctrl/internal/mapper/helm.go

```diff
@@ -71,17 +71,45 @@ func HelmSchemaToFields(name string, schema helm.Property, dependencies []*model
 }
 
 func sortFields(fields []models.Field, order []string) []models.Field {
-	ordersMap := make(map[string]int)
-
-	for i, s := range order {
-		ordersMap[s] = i
-	}
-
-	sort.Slice(fields, func(i, j int) bool {
-		return ordersMap[fields[i].Name] < ordersMap[fields[j].Name]
-	})
+    // If no custom order is provided, sort fields alphabetically by name
+    if len(order) == 0 {
+        // Extract names of fields for sorting
+        fieldNames := make([]string, len(fields))
+        for i, field := range fields {
+            fieldNames[i] = field.Name
+        }
+
+        // Sort field names alphabetically
+        sort.Strings(fieldNames)
+
+        // Map field names to their indices in the sorted array
+        fieldIndices := make(map[string]int)
+        for i, name := range fieldNames {
+            fieldIndices[name] = i
+        }
+
+        // Use the sorted indices to sort the fields
+        sort.Slice(fields, func(i, j int) bool {
+            return fieldIndices[fields[i].Name] < fieldIndices[fields[j].Name]
+        })
+    } else {
+        // Custom order provided, sort fields based on the order
+        ordersMap := make(map[string]int)
+
+        // Map field names to their indices in the custom order
+        for i, s := range order {
+            ordersMap[s] = i
+        }
+
+        // Sort fields based on their indices in the custom order
+        sort.Slice(fields, func(i, j int) bool {
+            return ordersMap[fields[i].Name] < ordersMap[fields[j].Name]
+        })
```

Comment:
Your solution works when the order is provided for all fields or not provided at all, but it doesn't work when 3/6 fields have the order defined. You can check the issue for further explanation of such a case.
I tried it with the following template, and I got the order from the picture.
```
repo: https://github.com/cyclops-ui/templates
path: demo-extended
version: field-alphabetical-sort
```
<img width="894" alt="Screenshot 2024-04-28 at 19 07 10" src="https://github.com/cyclops-ui/cyclops/assets/72022639/0f7893e9-52ff-4c5a-a834-2680e7bc46e9">